staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -74,6 +76,9 @@ var (
 	// We try to spread the load on apiserver by setting timeouts for
 	// watch requests - it is random in [minWatchTimeout, 2*minWatchTimeout].
 	minWatchTimeout = 5 * time.Minute
+
+	// Gather initial and resync lists in chunks of this size.
+	listPageSize = int64(10000)
```

Comment:
Experimentally chosen?  When I did the first round 500 worked ok for clients.

Reply:
We did some etcd range read experiments and found this to be a good number for watch cache to etcd traffic. From API clients we should probably stick with the 500 you previously found works well. I'll update this PR to only use 10k for watch cache to etcd.

Reply:
I don't understand why those are supposed to be different - this is the same thing, isn't it?

Reply:
I've updated the PR, maybe that will clarify?  I'm viewing them as:
1. watchcache->etcd list requests (proposing we go from full list to 10k item chunks)
2. clients using client reflector->apiserver->etcd list requests (proposing we go from full list to 500 item chunks)

Reply:
clients -> apiserver is more about transfer bandwidth (today) until we get gzip on by default.  With gzip on by default I think it's likely that we could bump the chunk size up from the client.

Reply:
OK - now I understand. Thanks.

> transfer bandwidth (today) until we get gzip on by default

I guess it depends on environment? We didn't see real issues with bandwidth in gcp.

Reply:
And note that the client reflector->apiserver->etcd list requests are only in 500 item chunks if the request is not served from watch cache, if it is, it remains a full list.